File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxAscendingSum(nums) {

         
        let res = 0;
        let count = 0 ;
        let prev = -1;

        nums.forEach(n => {
            if(n > prev){
                count += n;
                res = Math.max(count, res);
                prev = n;
            } else {
                count = n;
                prev = n;
            }

        })

        return res;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.java
class Solution {
    public int findDuplicate(int[] nums) {
        
        for(int i = 0 ; i < nums.length ; i++){
            int num = Math.abs(nums[i]) - 1;

            if(nums[num] < 0) return num + 1;

            else {
                nums[num] *= -1; 
            }
        }

        return -1;
    }
}
